offline/offline_preprocessing.py

```diff
@@ -0,0 +1,88 @@
+# %%
+'''
+Preprocessing of the P300 data.
+
+Author:
+    Karahan Yilmazer - 16.04.2022
+    Nathan van Beelen
+    Ayça Kepçe
+
+'''
+
+#!%matplotlib qt
+
+import glob
+import mne
+
+import pickle
+from utils import load_xdf_data
+
+
+
+# %%
+# Load the configurations of the paradigm
+with open(r'./config_durations.pkl', 'rb') as file:
+    exp_durations = pickle.load(file)
+
+#file = r"r'C:\Users\AYCA\PycharmProjects\biosignal_stars\data\\raw\\**\\*.xdf"
+file = r'C:\Users\AYCA\PycharmProjects\biosignal_stars\data\raw\*.xdf'
+
+epochs_list = []
+for trial_no, trial in enumerate(sorted(glob.glob(file))):
+    print("TRIAL NO ", trial_no)
+    raw, event_arr, event_id = load_xdf_data(file= trial)
+
+    # Implement the band-pass filter
+    flow, fhigh = 0.1, 10
+    raw_filt = raw.filter(flow, fhigh)
+    
+    # Apply Common Average Referencing.
+    raw_car, _ = mne.set_eeg_reference(raw_filt, 'average', copy = True, ch_type = 'eeg')
+
+    # Now we want to epoch our data to trials and do a baseline correction.
+    total_trial_duration = exp_durations['baseline_length'] + exp_durations['target_length'] + \
+                           exp_durations['highlight_length'] * exp_durations['num_highlights'] + \
+                           exp_durations['inter_highlight_length'] * (exp_durations['num_highlights'] - 1)
+    tmin, tmax = -exp_durations['baseline_length'], total_trial_duration
+    epochs_list.append(mne.Epochs(raw_car,
+                        events=event_arr,
+                        event_id=event_id,
+                        tmin=tmin,
+                        tmax=tmax,
+                        baseline=(-exp_durations['baseline_length'], 0), # We apply baseline correction here !!!!!!!!!!!!!
+                        preload=True,
+                        event_repeated='drop'))
+
+# Concatenate the epochs
+epochs = mne.concatenate_epochs(epochs_list)
+
+
+# Get a subselection of epochs that resembles the trials
+sub_epochs_trials = epochs['trial_begin']
+
+# Now we epoch the trials to get highlights
+tmin, tmax = 0, exp_durations['highlight_length']
```

Comment:
I think tmax should be independent of the highlight length. Since the length of the ERP is also independent of the highlight length.

Reply:
It was temporary, @MariyanaCh changed it to a proper value after literature review.